src/ISDevice.cpp

```diff
@@ -994,23 +1148,53 @@ int ISDevice::onIsbDataHandler(p_data_t* data, port_handle_t port)
     return 1;   // allow others to continue to process this message
 }
 
+int ISDevice::onIsbAckHandler(p_ack_t* ack, unsigned char packetIdentifier, port_handle_t port){
+
+    return 1;   // allow others to continue to process this message
+}
+
 // return 0 on success, -1 on failure
 int ISDevice::onNmeaHandler(const unsigned char* msg, int msgSize, port_handle_t port) {
     std::lock_guard<std::recursive_mutex> lock(portMutex);
 
     switch (getNmeaMsgId(msg, msgSize))
     {
         case NMEA_MSG_ID_INFO:
-            nmea_parse_info(devInfo, (const char*)msg, msgSize);
+        {	// IMX device Info
+            dev_info_t info;
+            nmea_parse_info(info, (const char*)msg, msgSize);
             hdwId = ENCODE_DEV_INFO_TO_HDW_ID(devInfo);
```

Comment:
This should ENCODE_DEV_INFO_TO_HDW_ID(info) --- not devInfo
The hdwRunState should also be made to info...

Or, these two lines need to be moved below the switch on the hardware type, otherwise, they will be overwritten by info.